Model/LdapUser.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+namespace FR3D\LdapBundle\Model;
+
+use Symfony\Component\Security\Core\User\AdvancedUserInterface;
+
+/**
+ * Simple user object.
+ */
+class LdapUser implements LdapUserInterface, AdvancedUserInterface
+{
+    const ROLE_DEFAULT = 'ROLE_USER';
+    const ROLE_SUPER_ADMIN = 'ROLE_SUPER_ADMIN';
+
+    /**
+     * @var string
+     */
+    protected $dn;
+
+    /**
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * @var string
+     */
+    protected $usernameCanonical;
+
+    /**
+     * @var string
+     */
+    protected $email;
+
+    /**
+     * @var string
+     */
+    protected $emailCanonical;
+
+    /**
+     * @var bool
+     */
+    protected $enabled;
+    /**
+     * The salt to use for hashing.
+     *
+     * @var string
+     */
+    protected $salt;
+
+    /**
+     * Encrypted password. Must be persisted.
+     *
+     * @var string
+     */
+    protected $password;
+
+    /**
+     * @var bool
+     */
+    protected $locked;
+    /**
+     * @var bool
+     */
+    protected $expired;
+
+    /**
+     * @var array
+     */
+    protected $roles;
+
+    /**
+     * @var bool
+     */
+    protected $credentialsExpired;
+
+    public function __construct()
+    {
+        $this->salt = base_convert(sha1(uniqid(mt_rand(), true)), 16, 36);
+        $this->enabled = false;
+        $this->locked = false;
+        $this->expired = false;
+        $this->roles = array();
+        $this->credentialsExpired = false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getDn()
+    {
+        return $this->dn;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setDn($dn)
+    {
+        $this->dn = $dn;
+    }
+
+    public function addRole($role)
```

Comment:
I am currently not sure how this should look like!? Do mean the addRole method only? Or everything related to role -> add/remove/has/set